prototype2/gdgem/vmm3/CalibrationFile.cpp

```diff
@@ -0,0 +1,89 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Class for handling calibration files for VMM asics
+///
+//===----------------------------------------------------------------------===//
+
+#include <common/Log.h>
+#include <common/Trace.h>
+#include <gdgem/vmm3/CalibrationFile.h>
+#include <nlohmann/json.hpp>
+#include <fstream>
+
+using json = nlohmann::json;
+
+// #undef TRC_LEVEL
+// #define TRC_LEVEL TRC_L_DEB
+
+
+/// \brief clear the calibration array
+CalibrationFile::CalibrationFile() {
+  auto n = sizeof(calibrations)/sizeof(calibration);
+  for (size_t i = 0; i < n; i++) {
+    ((calibration *)calibrations)[i] = nocorr;
```

Comment:
Though I personally also find them to be a bit of a pain, we should try to move a away from c-style casts and start using the safer C++11 casts (in this case `reinterpret_cast<>()`).

Reply:
I prefer readability and this is a pretty clean example ;-)